File: example/iOS/AppRevealExample/Services/ExampleStateContainer.swift
import Foundation

#if DEBUG
import AppReveal
#endif

class ExampleStateContainer {

    static let shared = ExampleStateContainer()

    var isLoggedIn: Bool = false
    var userEmail: String = ""
    var userName: String = "Test User"
    var selectedTab: Int = 0
    var cartItemCount: Int = 2
    var lastSyncDate: Date = Date()

    private init() {}
}

#if DEBUG
extension ExampleStateContainer: StateProviding {
    func snapshot() -> [String: AnyCodable] {
        [
            "isLoggedIn": AnyCodable(isLoggedIn),
            "userEmail": AnyCodable(userEmail),
            "userName": AnyCodable(userName),
            "selectedTab": AnyCodable(selectedTab),
            "cartItemCount": AnyCodable(cartItemCount),
            "lastSyncDate": AnyCodable(lastSyncDate.ISO8601Format()),
        ]
    }
}
#endif

File: Flutter/example/lib/services/example_state_container.dart
import 'package:appreveal/appreveal.dart';

class ExampleStateContainer implements StateProviding {
  static final instance = ExampleStateContainer._();
  ExampleStateContainer._();

  bool isLoggedIn = false;
  String userEmail = '';
  String userName = '';
  String selectedTab = 'catalog';
  int cartItemCount = 0;
  DateTime? lastSyncDate;

  @override
  Map<String, dynamic> snapshot() => {
    'isLoggedIn': isLoggedIn,
    'userEmail': userEmail,
    'userName': userName,
    'selectedTab': selectedTab,
    'cartItemCount': cartItemCount,
    'lastSyncDate': lastSyncDate?.toIso8601String(),
  };
}
